US19 - Add missing IEEE functions (#363)
j3-fortran/fortran_proposals: proposals/us19_ieee_funcs/draft-paper.txt

To: J3                                                     J3/26-###
From: Lorri Menard & JoR
Subject: US19 - Missing IEEE functions
Date: 2026-February-

> Reference: 23-111r1, 23-234r2, 25-143, 25-190r3, 25-007r1

Author: Suggest an explicit reference to IEEE-754, and removal of references not actually cited in the body of the paper.

```suggestion
Reference: 23-234r2, 25-143, 25-007r1, ISO/IEC/IEEE 60559:2020 ("IEEE-754")
```

  not be -1.
  <<Result characteristics.>> Same as X.
  <<Result value.>> The result has a value equal to a processor-dependent
  approximation to LOG2(X+1).

  It is recommended that the processor compute the result using a method
  that does not incur loss of precision by forming X+1.

  <<Example.>> LOG2P1(3.0) has the value 2.0 (approximately).

[470:21+] insert one intrinsic {note: after RESHAPE}

16.9.175+ ROOTN(X,N)
  <<Description.>> Radical function.
  <<Class.>> Elemental function.
  <<Arguments.>> X shall be of type real or complex.
                 N shall be of type integer, and shall not be zero.
  <<Result characteristics.>> Same as X.
  <<Result value.>> The result has a value equal to a processor-dependent
  approximation to X**(1./N).  A result of type complex is the principal
  value with the real part greater than or equal to zero.  When the
  real part of the result is zero, the imaginary part has the same sign
  as the imaginary part of X.

  <<Example>> ROOTN(27.0, 3) has the value 3.0 (approximately).

[470:31+] insert one intrinsic {note: after RRSPACING}

16.9.176+ RSQRT(X)
  <<Description.>> Computes the reciprocal of the square root of a

>   given value.

Author: Adjust to match the description proposed for Table 16.1

```suggestion
  <<Description.>> Reciprocal square root.
```

  <<Class.>> Elemental function
  <<Argument.>> X shall be of type real or complex.  If X is real its

>   value shall be greater than or equal to zero.  If X is complex its

Author: X equal to zero implies a divide-by-zero, and should be prohibited

```suggestion
  <<Argument.>> X shall be of type real or complex.  If X is real its
  value shall be greater than zero.  If X is complex its
```
